### ISSUE TITLE
Driver name mismatch for Apple M1, M1 Pro & M1 Max

### ISSUE THREAD

**Issue by Pwuts** [SergeyPirogov/webdriver_manager#506]
* https://github.com/SeleniumHQ/selenium/issues/11066
* https://github.com/Significant-Gravitas/Auto-GPT/issues/2600

Example fix from Ruby webdriver manager:
* https://github.com/titusfortner/webdrivers/pull/239/files

**Comment by aleksandr-kotlyar** [comment]
Already has this here
https://github.com/SergeyPirogov/webdriver_manager/blob/master/webdriver_manager/drivers/chrome.py#L46

**Comment by Pwuts** [comment]
@aleksandr-kotlyar example log is in the OP of https://github.com/Significant-Gravitas/Auto-GPT/issues/2600.
```
====== WebDriver manager ======
Current google-chrome version is 112.0.5615
Get LATEST chromedriver version for 112.0.5615 google-chrome
There is no [mac64_m1] chromedriver for browser 112.0.5615 in cache
Trying to download new driver from https://chromedriver.storage.googleapis.com/112.0.5615.49/chromedriver_mac64_m1.zip
SYSTEM:  Command browse_website returned: Error: There is no such driver by url https://chromedriver.storage.googleapis.com/112.0.5615.49/chromedriver_mac64_m1.zip
 THOUGHTS:  It seems like we need to use a web driver to browse the website. I think we should use Selenium to automate the browsing process and extract the information we need.
REASONING:  Selenium is a powerful tool for automating web browsing and can be used to extract information from websites. By using Selenium, we can automate the browsing process and extract the information we need without having to manually navigate the website.
```

Driver version to download: ``  
Logic:
```py
# driver_version_to_download = '112.0.5615.49'

if version.parse(driver_version_to_download) < version.parse("106.0.5249.61"):	# false
    os_type = os_type.replace("mac_arm64", "mac64_m1")
```
Expected output: `mac_arm64`

**Comment by Pwuts** [comment]
@aleksandr-kotlyar do you still need more info?

**Comment by aleksandr-kotlyar** [comment]
@Pwuts thank you! Still need package version,  actual os architecture, and expected behavio for each architecture in topic: M1, M1 PRO, M1 MAX or link to docs where describes what pattern for each of it should be below 106, after 106 and other versions of chrome.

**Comment by yogendrasinghx** [comment]
@Pwuts I had experienced a similar issue on my Mac Mini M1 with Chrome version 112.0.5615.137 (Official Build) (arm64), but I was able to resolve it by updating my version of webdriver-manager. Specifically, after updating to version 3.8.6, my program was able to run without any further issues.

Based on my experience, I would suggest that you try upgrading to the latest version of the webdriver-manager package. Doing so may help to resolve the issue you are experiencing.

Best regards
